### PR TITLE
Tests: Post Update Review

### PR DESCRIPTION
**Pull request by epreston**
Related issue: none.

**Description**

This cleans up the unit tests to make them more consistent. Use Qunit hooks to provision and cleanup global mocks in Audio. Explicit block scope for variables used in tests.

### CODE REVIEW

File: test/unit/src/audio/AudioContext.tests.js [hunk 1]
```diff
@@ -4,29 +4,41 @@ import { AudioContext } from '../../../../src/audio/AudioContext.js';
...
 
-					},
-				};
+		}
+
+		if ( typeof window === 'undefined' ) {
+
+			hooks.before( function () {
+
+				mockWindowAudioContext();
```
Comment (LeviPesin) [inline test/unit/src/audio/AudioContext.tests.js:33]: Shouldn't you then remove explicit calls to this function later, e.g. on line 48?

Comment (epreston) [reply test/unit/src/audio/AudioContext.tests.js:33]: You mean where that `after` hook queues to clean up from this function 2 lines below ?

**Comment by epreston** [comment]
I've made a bunch of notes.  One on this topic is about moving to vite / vitest.  I'm sure you understand the benefits. Happy to reformat the tests in jest / vitest compatible units.

The recent changes in the build and dropping babel (to let the final app bundler do the translation) means that this project nearly has a default vite config for a lib.  Its fast, nice to have live test feedback as you develop (unit tests, preview, type checking, linting, vscode integration). Might pare nicely with the e2e puppeteer stuff.

I'm not sure where these conversations happen.  I've got a list of notes, questions, options.  Not too fixed on any one direction except for filling in these unit tests while updating the documentation.  Point me where to write them out.

**Comment by LeviPesin** [comment]
Maybe you can create a new issue and write them there?

**Comment by epreston** [comment]
@mrdoob I'd like to get back to filling in tests.  Is there anything I can do to break this up for review ?

**Pull request merged**